### PR TITLE
Update setup.py: update TRT version

### PR DESCRIPTION
**Pull request by snnn**
### Description
As a follow-up of #20506

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -230,7 +230,7 @@ def run(self):
                     "libmigraphx_tf.so.2",
                 ]
 
-                tensorrt_dependencies = ["libnvinfer.so.8", "libnvinfer_plugin.so.8", "libnvonnxparser.so.8"]
+                tensorrt_dependencies = ["libnvinfer.so.10", "libnvinfer_plugin.so.10", "libnvonnxparser.so.10"]
```
Comment (jywu-msft) [inline setup.py:233]: +@yifanl fyi

Comment (jywu-msft) [reply setup.py:233]: do we need this? is there a symlink from lbnvinfer.so to libnvinfer.so.8 ? would we be able add libnvinfer.so ?

Comment (jywu-msft) [reply setup.py:233]: did you actually hit an issue with this?
i'm actually not sure we need this stuff anymore. wasn't it used in the old days to work around auditwheel and ensure it doesn't try to copy these dependencies into the whl?

Comment (yf711) [reply setup.py:233]: Each TRT does have a symlink:
i.e For TRT8.6 installed in Ubuntu:
image
After installing TRT10, libnvinfer.so was re-linked to libnvinfer.so.10:
image

Comment (jywu-msft) [reply setup.py:233]: ok this change is required. i downloaded whl package and see the libnvinfer libs copied in there under ort-nightly-gpu.libs

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**